onnxruntime/python/session.py

```diff
@@ -19,28 +20,11 @@ class Session:
     """
     This is the main class used to run a model.
     """
-    def __init__(self, sess):
-        self._enable_fallback = True
+    def __init__(self):
 
-    def _reset_session(self):
-        "release underlying session object."
```

Comment:
Moved this down to InferenceSession given all the implementation logic is specific to that derived class, and the recreation of the C.InferenceSession is done by bindings in that derived class.

We could possibly merge the python Session and InferenceSession classes and remove a lot of the confusing aspects of this setup.